tests/test_views.py

```diff
@@ -0,0 +1,137 @@
+            password='blablabla',
+            is_superuser=True, is_staff=True,
+            email='testuser@gmail.com',
+            is_active=True,
+        )
+
+    def test_index_view_rendering(self):
+        """Testing main page for render needed posts."""
+        time_zone = timezone.get_current_timezone()
+        future_post = Post.objects.create(
+            author=self.user,
+            title='future_test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1,
+                month=4,
+                year=3020,
+                tzinfo=time_zone,
+            ),
+        )
+        post = Post.objects.create(
+            author=self.user, title='Test',
+            text='superText',
+            created_date=datetime(
+                day=1, month=3,
+                year=2020, tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=3,
+                year=2020,
+                tzinfo=time_zone,
+            ),
+        )
+        past_post = Post.objects.create(
+            author=self.user, title='past_est',
+            text='superText',
+            created_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+            published_date=datetime(
+                day=1, month=4,
+                year=2019,
+                tzinfo=time_zone,
+            ),
+        )
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=1,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            for _ in response:
+                with self.subTest():
+                    self.assertListEqual(
+                        list(response.context['posts']), [
+                            past_post,
+                        ],
+                    )
+                    self.assertNotContains(response, post)
+                    self.assertNotContains(response, future_post)
+        with patch(
+                'django.utils.timezone.now', lambda: datetime(
+                    day=1,
+                    month=10,
+                    year=2020,
+                    tzinfo=time_zone,
+                ),
+        ):
+            response = self.client.get(reverse('post_list'))
+            for _ in response:
+                with self.subTest():
+                    self.assertListEqual(
+                        list(response.context['posts']), [
+                            post,
+                            past_post,
+                        ],
+                    )
+                    self.assertNotContains(response, future_post)
```

Comment:
You just repeat exactly the same test in a loop here. Doing the same thing over and over isn't gonna influence the result.

Reply:
Thanks. Can you show in more detail what I have to do?

Reply:
You need to have some iterable (list or tuple, or even dict) containing the test cases (with the current date in different ranges) mapped to the expected rendered list.
Then, you'd iterate over the cases in a loop, on each iteration open a context manager for subtest, patch the date and test the expected list. It's also useful to set a param for `self.subTest()`.